documentation/api/swagger-spec.rst

```diff
@@ -0,0 +1,59 @@
+.. _swagger:
+
+Generate Swagger specification file for server RESTful APIs
+######################
+
+This section will introduce how to set up swagger specification auto-generation tool `swagger-autogen <https://github.com/swagger-autogen/swagger-autogen>`_, how to use it to generate spec file and how to visualize and try the requests execution with `Swagger UI
+<https://github.com/swagger-api/swagger-ui>`_ tool.
+
+Generation
+**********
+
+In ``package.json``, ensure the ``swagger-autogen`` npm package definition has been added in the``devDependencies`` field, and the ``swagger-autogen`` script command has been added in the ``scripts`` field. Besides, the ``swagger.js`` file in root is the key configuration file of the ``swagger-autogen`` module, including the target file, endpoints source file and the schema definitions. More details are available `here
+<https://github.com/swagger-autogen/swagger-autogen#usage-with-optionals>`_.
+
+.. code-block:: shell
+
+   $ npm install # to install the dependencies.
+   $ npm run swagger-autogen # to generate the swagger specification file
+
+The ``swagger-output.json`` under the project root directory will be generated after running above commands.
+
+Visualization & Interaction
+***************************
+
+In swagger specification file, you can browse the definitions, request parameters, request payload, response payload and their corresponding examples of all APIs. Moreover, with the help of swagger UI tool, you can visualize and interact with the API's resources without having any of the implementation logic in place.
+
+There are many useful tools available to achieve the functionalities mentioned above.
+
+If you are using `VS Code`, you can install the ``Swagger Viewer`` (ADD a link here) from the extension market, and execute the ``Preview Swagger`` command from the `swagger specification` file editor to visualize and interact the API's resource as following images shows.
```

Comment:
- remove `ADD a link here`  and add an actual link.
- add `with` after interact: interact `with` the API's resource